test-data/run.test

```diff
@@ -2385,3 +2385,63 @@ g()
 [out]
 (((1, 2), 2), 1, 'hi', 'hi2')
 caught the exception
+
+[case testYield]
+from typing import Generator
+def yield_twice_and_return() -> Generator:
+    yield 1
+    yield 2
+    return 4
+    yield 3
+
+# def yield_loop() -> Generator:
+#     i = 0
+#     while i < 5:
+#         if i == 3:
+#             return i
+#         yield i
+#         i += 1
```

Comment:
Why is this all commented out?

Also, we can give better types than just `Generator` to our generators. See https://mypy.readthedocs.io/en/latest/kinds_of_types.html#generators